doc/WAF.md

```diff
@@ -19,17 +21,28 @@ appSync:
       - disableIntrospection
 ```
 
+Or directly associate an existing web ACL:
+
+```yaml
+appSync:
+  name: my-api
+  waf:
+    enabled: true
+    arn: 'arn:aws:wafv2:us-east-1:123456789012:regional/webacl/my-Waf/d7b694d2-4f7d-4dd6-a9a9-843dd1931330'
+```
+
 ## Configuration
 
 - `enabled`: Boolean. Enable or disable WAF. Defaults to `true` when `appSync.waf` is defined.
+- `arn`: Optional. The WAF's ARN to associate with your AppSync resource.
 - `name`: Optional. The name of this WAF instance. Defaults to the name of your API.
 - `defaultAction`: Optional. The default action if a request does not match a rule. `Allow` or `Block`. Defaults to `Allow`.
-- `description`: A description for this WAF instance.
+- `description`: Optional. A description for this WAF instance.
 - `visibilityConfig`: Optional. A [visibility config](https://docs.aws.amazon.com/waf/latest/APIReference/API_VisibilityConfig.html) for this WAF
   - `name`: Metric name
   - `cloudWatchMetricsEnabled`: A boolean indicating whether the associated resource sends metrics to Amazon CloudWatch
   - `sampledRequestsEnabled`: A boolean indicating whether AWS WAF should store a sampling of the web requests that match the rule
-- `rules`: An array of [rules](#rules).
+- `rules`: Required. An array of [rules](#rules). Optional when `arn` is present
```

Comment:
This is a bit confusing. 
It might let user think that you can pass new `rules` even if you pass a waWAF `arn`, which is not from what I can see in the code (the WAF is only associated).
(same for the other attributes).

This is probably fine for now. I'd like to reorganize the whole documentation anyway.